webapp/src/app/routes/yaml/yaml.service.ts

```diff
@@ -93,13 +93,17 @@ export class YamlService {
           const switchBlock = block as SwitchBlock;
           const cases: SwitchCase[] = [];
           for (const blockCase of switchBlock.cases) {
-            const switchCase: SwitchCase = {
-              steps: this.blocksToYaml(blockCase.steps).steps
-            };
+            let switchCase: SwitchCase;
             if (blockCase.isDefault) {
-              switchCase.default = blockCase.isDefault;
+              switchCase = {
+                default: blockCase.isDefault,
+                steps: this.blocksToYaml(blockCase.steps).steps,
+              };
             } else {
-              switchCase.match = blockCase.match;
+              switchCase = {
+                match: blockCase.match,
+                steps: this.blocksToYaml(blockCase.steps).steps,
+              };
```

Comment:
How about putting it once, after the `if`?

`switchCase.steps = this.blocksToYaml(blockCase.steps).steps;`

Reply:
@oferb, 
Problem is steps is required field, so we can't create case without steps.
We could make steps non required, but it's kind of code crutch, because actually case always should have steps.
Also, steps is complex type, and if we make steps non required for switch case, we also make it non required for all other types with steps.